modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/properties/time/TimestampFileNameExtractor.java

```diff
@@ -58,13 +77,30 @@ public void setProperties(SimpleFeature feature) {
         final List<Date> dates = new ArrayList<Date>();
         for (String match : getMatches()) {
             // try to convert to date
+
             try {
                 if (customFormat != null) {
                     Date parsed = customFormat.parse(match);
+                    if (inclusive) {
+                        // Special case to make t and z upper case for ISO8601 matching.
+                        format = format.replace("t", "T").replace("z", "Z");
```

Comment:
Is this part of the refactor, seems some new functionality?

Reply:
That was part of the improval of the ImageMosaic Time parser to support the "fuzzy Date search" available in GeoServer's TimeParser which returns the end of a timeRange interval on a reduced accuracy time.
such as a reduced accuracy datetime "2019-11" get translated into the time range 2019-11-01T00:00:00.000 to 2019-11-30T23.59.59.999